src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -120,14 +127,14 @@ def _check_pid_changed(self):
   def _handle_batch_event(self, files):
     self._logger.debug('handling change event for: %s', files)
 
-    self._maybe_invalidate_scheduler_batch(files)
-
     invalidated = self._scheduler.invalidate_files(files)
     if invalidated:
       self._loop_condition.notify_all()
 
+    self._maybe_invalidate_scheduler_batch(files)
```

Comment:
The `Snapshot` that we're computing should always be for the same set of `globs`: `self._invalidation_globs`. So neither `self._maybe_invalidate_scheduler_batch` nor `self._get_snapshot` actually need to take arguments: they can just use `self._invalidation_globs` directly.